fhir-path/src/main/java/com/ibm/fhir/path/function/ResolveFunction.java

```diff
@@ -94,16 +112,51 @@ public int getMaxArity() {
                             }
                         }
                     } else {
-                        Matcher matcher = REFERENCE_PATTERN.matcher(referenceReference);
-                        if (matcher.matches()) {
-                            resourceType = matcher.group(RESOURCE_TYPE_GROUP);
-                            if (referenceType != null && !resourceType.equals(referenceType)) {
-                                throw new IllegalArgumentException("Resource type found in reference URL does not match reference type");
+                        if (isBundleContext) {
+                            // we know the root of the tree is a Bundle and the current node is of type Reference,
+                            // so walk up the tree until we get to highest Bundle.entry.fullUrl in the tree
+                            // and save the value of its fullUrl
```

Comment:
While the comment really does help... .

Could you layout a small example? or link to a test example? 

If I have to maintain it, it'd help to laser focus on the 'aaah-ha' that's what it does.

Reply:
I added some examples to this comment (and also a test case for the Bundle-in-Bundle case)